Reproducible Failure in CollapsingTopDocsCollectorSearchAfterTests.testCollapseDouble

Just ran into this locally on master and it reproduces 100% of the time:

```

> Task :server:test
OpenJDK 64-Bit Server VM warning: Option --illegal-access is deprecated and will be removed in a future release.

REPRODUCE WITH: ./gradlew ':server:test' --tests "org.apache.lucene.grouping.CollapsingTopDocsCollectorSearchAfterTests.testCollapseDouble" -Dtests.seed=57C720291214B3A3 -Dtests.locale=fr-LU -Dtests.timezone=America/Los_Angeles -Druntime.java=16

org.apache.lucene.grouping.CollapsingTopDocsCollectorSearchAfterTests > testCollapseDouble FAILED
    java.lang.AssertionError: expected:<1244> but was:<0>
        at __randomizedtesting.SeedInfo.seed([57C720291214B3A3:FFDF62F1CC5A52B3]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.failNotEquals(Assert.java:834)
        at org.junit.Assert.assertEquals(Assert.java:645)
        at org.junit.Assert.assertEquals(Assert.java:631)
        at org.apache.lucene.grouping.CollapsingTopDocsCollectorSearchAfterTests.assertSearchCollapse(CollapsingTopDocsCollectorSearchAfterTests.java:114)
        at org.apache.lucene.grouping.CollapsingTopDocsCollectorSearchAfterTests.assertSearchCollapse(CollapsingTopDocsCollectorSearchAfterTests.java:56)
        at org.apache.lucene.grouping.CollapsingTopDocsCollectorSearchAfterTests.testCollapseDouble(CollapsingTopDocsCollectorSearchAfterTests.java:230)

```

--- a/server/src/main/java/org/apache/lucene/search/grouping/CollapsingTopDocsCollector.java
+++ b/server/src/main/java/org/apache/lucene/search/grouping/CollapsingTopDocsCollector.java
@@ -69,7 +69,7 @@ public final class CollapsingTopDocsCollector<T> extends FirstPassGroupingCollec
     public CollapseTopFieldDocs getTopDocs() throws IOException {
         Collection<SearchGroup<T>> groups = super.getTopGroups(0);
         if (groups == null) {
-            TotalHits totalHits = new TotalHits(0, TotalHits.Relation.EQUAL_TO);
+            TotalHits totalHits = new TotalHits(totalHitCount, TotalHits.Relation.EQUAL_TO);
             return new CollapseTopFieldDocs(collapseField, totalHits, new ScoreDoc[0], sort.getSort(), new Object[0]);
         }
         FieldDoc[] docs = new FieldDoc[groups.size()];

--- a/server/src/test/java/org/apache/lucene/grouping/CollapsingTopDocsCollectorSearchAfterTests.java
+++ b/server/src/test/java/org/apache/lucene/grouping/CollapsingTopDocsCollectorSearchAfterTests.java
@@ -223,7 +223,7 @@ public class CollapsingTopDocsCollectorSearchAfterTests extends ESTestCase {
             @Override
             public SortField sortField(boolean reversed) {
                 SortField sortField = new SortField("field", SortField.Type.DOUBLE, reversed);
-                sortField.setMissingValue(reversed ? Double.MIN_VALUE : Double.MAX_VALUE);
+                sortField.setMissingValue(reversed ? Double.NEGATIVE_INFINITY : Double.POSITIVE_INFINITY);
                 return sortField;
             }
         };

